src/qibolab/backends.py

```diff
@@ -50,6 +52,43 @@ def __init__(self, platform):
         }
         self.compiler = Compiler.default()
 
+    @property
+    def qubits(self) -> list[Union[str, int]]:
+        """Returns the qubits in the platform."""
+        return list(self.platform.qubits)
+
+    @property
+    def connectivity(self) -> list[tuple[Union[str, int], Union[str, int]]]:
+        """Returns the list of connected qubits."""
+        return list(self.platform.pairs)
+
+    @property
+    def natives(self) -> list[str]:
+        """Returns the list of native gates supported by the platform."""
+        compiler = Compiler.default()
+        natives = [g.__name__ for g in list(compiler.rules)]
+
+        check_2q = ["CZ", "CNOT"]
+        for gate in check_2q:
+            if gate in natives:
+                for pair in self.connectivity:
+                    logical_pair = [list(self.platform.qubits).index(q) for q in pair]
+                    if self._is_gate_calibrated(
+                        getattr(gates, gate)(*logical_pair), compiler
+                    ):
+                        break
+                else:
+                    natives.remove(gate)
+        return natives
+
+    def _is_gate_calibrated(self, gate, compiler) -> bool:
+        """Helper method to check if a gate is calibrated."""
+        try:
+            compiler[type(gate)](gate, self.platform)
+            return True
+        except ValueError:
+            return False
```

Comment:
Yes, but this is for `0.2`. To create something similar to `natives`,

> Thus, you should not directly inspect the `self.platform.qubits[q].native_gates` attribute, but rather you should take them from here:

this will be done for pairs.

Reply:
> Yes, but this is for `0.2`. To create something similar to `natives`,

Ah, sorry, you're right. My mind is shifted on that side.
For 0.1, you just need to replace
```py
        calibrated = self.platform.natives.two_qubit
        ...
                calibrated[pair][gate] is None for pair in self.connectivity
```
with:
```py
        calibrated = self.platform.pairs
        ...
                calibrated[pair].native_gates[gate] is None for pair in self.connectivity
```

Of course, that's the general idea, it should be tested.

Reply:
> > Thus, you should not directly inspect the `self.platform.qubits[q].native_gates` attribute, but rather you should take them from here:
> 
> this will be done for pairs.

As I said above, I just changed my mind ^^

The hybrid approach is simpler (take all the single-qubit gates, and pre-filtered two-qubit ones, from the compiler - and then filter with direct inspection, which is what you would do anyhow, just indirectly.